esm/inverse_folding/gvp_transformer.py

```diff
@@ -130,6 +132,8 @@ def sample(self, coords, partial_seq=None, temperature=1.0, confidence=None):
             logits = logits[0].transpose(0, 1)
             logits /= temperature
             probs = F.softmax(logits, dim=-1)
+            if incremental_state is None:
```

Comment:
This will make it super slow right? Wouldn't it be better to just remove the `continue` at L126, and instead do something like

```
assert sampled_tokens.shape[0] == 1  # just listing an assumption here
if sampled_tokens[0, i] == mask_idx:
    sampled_tokens[:, i] = torch.multinomial(probs, 1).squeeze(-1)
```
at L137?

Reply:
Sorry didn't see this comment earlier. Thanks for catching that -- agreed and updated the code.